arangod/Agency/Supervision.cpp

```diff
@@ -669,8 +688,15 @@ std::vector<check_t> Supervision::check(std::string const& type) {
       if (transientHealthRecordFound) {
         // Calculate elapsed since lastAcked
         auto elapsed = std::chrono::duration<double>(startTimeLoop - lastAckedTime);
-
-        if (elapsed.count() <= _okThreshold) {
+        
+        // if there is an agreement on how to handle unhealthy servers,
+        // the "healthy" flag could be used here in the future to downgrade
+        // the health of servers that report back to us, but with an
+        // unhealthy status
+        // TODO: figure this out and add to design document - until then
+        // isHealthy remains unused here!
+
+        if (elapsed.count() <= _okThreshold /*&& isHealthy*/) {
```

Comment:
Even if we do not declare the server FAILED, we could still report any change in the `isHealthy` flag to the persisted Raft log under `/arango/Supervision/Health` and thus export it to the normal health check API. We would have to adjust `diffStat` in this file and write changes to Raft.